graphrag/index/flows/create_final_embeddings.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (c) 2024 Microsoft Corporation.
+# Licensed under the MIT License
+
+"""All the steps to transform the text units."""
+
+import logging
+from typing import cast
+
+import pandas as pd
+from datashaper import (
+    VerbCallbacks,
+)
+
+from graphrag.index.cache import PipelineCache
+from graphrag.index.config.embeddings import (
+    community_full_content_embedding,
+    community_summary_embedding,
+    community_title_embedding,
+    document_raw_content_embedding,
+    entity_description_embedding,
+    entity_name_embedding,
+    relationship_description_embedding,
+    text_unit_text_embedding,
+)
+from graphrag.index.operations.embed_text import embed_text
+from graphrag.index.operations.snapshot import snapshot
+from graphrag.index.storage import PipelineStorage
+
+log = logging.getLogger(__name__)
+
+
+async def create_final_embeddings(
+    final_documents: pd.DataFrame,
+    final_relationships: pd.DataFrame,
+    final_text_units: pd.DataFrame,
+    final_entities: pd.DataFrame,
+    final_community_reports: pd.DataFrame,
+    callbacks: VerbCallbacks,
+    cache: PipelineCache,
+    storage: PipelineStorage,
+    embedded_fields: set[str],
+    base_text_embed: dict | None = None,
+) -> pd.DataFrame:
+    """All the steps to generate all embeddings."""
+    documents_embeddings = final_documents[["id", "raw_content"]].copy()
```

Comment:
Do we need to copy these values? Can we just mantain references?

Reply:
If I don't copy them the workflow execution fails because python throws an error that the dataframe should be copied in order to manipulate it